src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/Utils.java

```diff
@@ -1553,15 +1555,16 @@ public String propertyName(ExecutableElement e) {
     }
 
     /**
-     * Returns true if the element is included, contains &#64;hidden tag,
+     * Returns true if the element is included or selected, contains &#64;hidden tag,
      * or if javafx flag is present and element contains &#64;treatAsPrivate
      * tag.
      * @param e the queried element
      * @return true if it exists, false otherwise
      */
     public boolean hasHiddenTag(Element e) {
-        // prevent needless tests on elements which are not included
-        if (!isIncluded(e)) {
+        // prevent needless tests on elements which are neither included nor selected.
+        // Non-included members may still be visible via "transclusion" from undocumented enclusure
+        if (!isIncluded(e) && !configuration.docEnv.isSelected(e)) {
```

Comment:
This comment applies to the whole of this method body.

I think the `@hidden` (or @treatAsprivate`) tag is sufficiently special that it should be handled separately, especially in the cases where the containing comment would not otherwise be read, because the element is not included/selected. See other/earlier comments about avoiding `getDocComment` and going to the underlying `getDocCommentInfo`, which will (should?) completely bypass `getDocComment` and (importantly) the use of doclint.